flink-connectors/flink-connector-pulsar/src/main/java/org/apache/flink/connector/pulsar/source/enumerator/PulsarSourceEnumerator.java

```diff
@@ -148,8 +162,42 @@ public void close() {
      * @return Set of subscribed {@link TopicPartition}s
      */
     private Set<TopicPartition> getSubscribedTopicPartitions() {
-        return subscriber.getSubscribedTopicPartitions(
-                pulsarAdmin, rangeGenerator, context.currentParallelism());
+        int parallelism = context.currentParallelism();
+        Set<TopicPartition> partitions =
+                subscriber.getSubscribedTopicPartitions(pulsarAdmin, rangeGenerator, parallelism);
+
+        // Seek start position for given partitions.
+        seekStartPosition(partitions);
```

Comment:
I think this should only be called in `assignPendingPartitionSplits`. Now if you have discover enabled, it would actually seek too often.